generator/transit_generator.cpp

```diff
@@ -229,52 +158,20 @@ void DeserializerFromJson::operator()(StopIdRanges & rs, char const * name)
   rs = StopIdRanges({stopIds});
 }
 
-void BuildTransit(string const & mwmDir, string const & countryId,
-                  string const & osmIdToFeatureIdsPath, string const & transitDir)
+// GraphData --------------------------------------------------------------------------------------
+void GraphData::DeserializeFromJson(my::Json const & root, OsmIdToFeatureIdsMap const & mapping)
 {
-  LOG(LERROR, ("This method is under construction and should not be used for building production mwm "
-      "sections."));
-  NOTIMPLEMENTED();
-
-  std::string const mwmPath = my::JoinFoldersToPath(mwmDir, countryId + DATA_FILE_EXTENSION);
-  string const graphFullPath = my::JoinFoldersToPath(transitDir, countryId + TRANSIT_FILE_EXTENSION);
-
-  Platform::EFileType fileType;
-  Platform::EError const errCode = Platform::GetFileType(graphFullPath, fileType);
-  if (errCode != Platform::EError::ERR_OK || fileType != Platform::EFileType::FILE_TYPE_REGULAR)
-  {
-    LOG(LINFO, ("For mwm:", mwmPath, TRANSIT_FILE_EXTENSION, "file not found"));
-    return;
-  }
-
-  // @TODO(bykoianko) In the future transit edges which cross mwm border will be split in the generator. Then
-  // routing will support cross mwm transit routing. In current version every json with transit graph
-  // should have a special name: <country id>.transit.json.
-
-  LOG(LINFO, (TRANSIT_FILE_TAG, "section is being created. Country id:", countryId, ". Based on:", graphFullPath));
-
-  string jsonBuffer;
-  try
-  {
-    GetPlatform().GetReader(graphFullPath)->ReadAsString(jsonBuffer);
-  }
-  catch (RootException const & ex)
-  {
-    LOG(LCRITICAL, ("Can't open", graphFullPath, ex.what()));
-  }
-
-  my::Json root(jsonBuffer.c_str());
-  CHECK(root.get() != nullptr, ("Cannot parse the json file:", graphFullPath));
-
-  OsmIdToFeatureIdsMap mapping;
-  FillOsmIdToFeatureIdsMap(osmIdToFeatureIdsPath, mapping);
-
-  // Note. |gates| has to be deserialized from json before starting writing transit section to mwm since
-  // the mwm is used to filled |gates|.
-  vector<Gate> gates;
-  DeserializeGatesFromJson(root, mwmDir, countryId, mapping, gates);
-  CHECK(IsValid(gates), (gates));
+  DeserializeItemFromJson(root, "stops", mapping, m_stops);
```

Comment:
Is it possible to apply Visitor-s approach to GraphData too?

Reply:
Yes. It's already done.